File: android-app/app/src/main/java/com/lifeplus/healthcare/network/ApiService.java
package com.lifeplus.healthcare.network;

import com.lifeplus.healthcare.model.*;
import java.util.List;
import java.util.Map;
import retrofit2.Call;
import retrofit2.http.*;

public interface ApiService {

    // ── App Settings & Slides (public — no auth needed) ───────────────────────
    @GET("settings")       Call<List<Map<String, Object>>> getSettings();
    @GET("settings/{key}") Call<Map<String, Object>>       getSetting(@Path("key") String key);
    @GET("slides")         Call<List<DashboardSlide>>      getDashboardSlides();
    @GET("walkthrough")    Call<List<WalkthroughSlide>>    getWalkthroughSlides();

    // ── Auth ──────────────────────────────────────────────────────────────────
    @POST("auth/login")    Call<AuthModels.AuthResponse> login(@Body AuthModels.LoginRequest b);
    @POST("auth/register") Call<AuthModels.AuthResponse> register(@Body AuthModels.RegisterRequest b);
    @POST("auth/refresh")  Call<AuthModels.AuthResponse> refresh(@Body AuthModels.RefreshRequest b);

    // ── Users ─────────────────────────────────────────────────────────────────
    @GET("users/me")       Call<Map<String, Object>> getMe();
    @PUT("users/{id}")     Call<Map<String, Object>> updateUser(@Path("id") long id, @Body Map<String, String> body);

    // ── Donors ────────────────────────────────────────────────────────────────
    @GET("donors")         Call<List<Donor>> getDonors();
    @GET("donors/{id}")    Call<Donor>       getDonor(@Path("id") long id);
    @GET("donors/search")  Call<List<Donor>> searchDonors(@Query("bloodGroup") String bg, @Query("district") String d);
    @GET("donors/my")      Call<List<Donor>> myDonors();
    @GET("donors/{id}/eligibility") Call<Map<String, Object>> checkEligibility(@Path("id") long id);
    @POST("donors")        Call<Donor>       createDonor(@Body Donor donor);
    @PUT("donors/{id}")    Call<Donor>       updateDonor(@Path("id") long id, @Body Donor donor);
    @DELETE("donors/{id}") Call<Void>        deleteDonor(@Path("id") long id);

    // ── Blood Requests ────────────────────────────────────────────────────────
    @GET("blood-requests")         Call<List<BloodRequest>> getBloodRequests();
    @GET("blood-requests/{id}")    Call<BloodRequest>       getBloodRequest(@Path("id") long id);
    @GET("blood-requests/search")  Call<List<BloodRequest>> searchBloodRequests(@Query("bloodGroup") String bg, @Query("district") String d);
    @GET("blood-requests/my")      Call<List<BloodRequest>> myBloodRequests();
    @GET("blood-requests/for-donor") Call<List<BloodRequest>> getBloodRequestsForDonor();
    @POST("blood-requests")        Call<BloodRequest>       createBloodRequest(@Body BloodRequest r);
    @PUT("blood-requests/{id}")    Call<BloodRequest>       updateBloodRequest(@Path("id") long id, @Body BloodRequest r);
    @PUT("blood-requests/{id}/status") Call<BloodRequest>    updateBloodRequestStatus(@Path("id") long id, @Query("status") String status);
    @DELETE("blood-requests/{id}") Call<Void>               deleteBloodRequest(@Path("id") long id);

    // ── Emergency Requests ────────────────────────────────────────────────────
    @GET("emergency-requests")             Call<List<EmergencyRequest>> getEmergencyRequests();
    @GET("emergency-requests/my")          Call<List<EmergencyRequest>> myEmergencyRequests();
    @POST("emergency-requests")            Call<EmergencyRequest>       createEmergencyRequest(@Body EmergencyRequest r);
    @PUT("emergency-requests/{id}/status") Call<EmergencyRequest>       updateEmergencyStatus(@Path("id") long id, @Query("status") String status);

    // ── Hospitals ─────────────────────────────────────────────────────────────
    @GET("hospitals")          Call<List<Hospital>> getHospitals();
    @GET("hospitals/{id}")     Call<Hospital>       getHospital(@Path("id") long id);
    @GET("hospitals/search")   Call<List<Hospital>> searchHospitals(@Query("district") String d);
    @POST("hospitals")         Call<Hospital>       createHospital(@Body Hospital h);
    @PUT("hospitals/{id}")     Call<Hospital>       updateHospital(@Path("id") long id, @Body Hospital h);
    @DELETE("hospitals/{id}")  Call<Void>           deleteHospital(@Path("id") long id);

    // ── Clinics ───────────────────────────────────────────────────────────────
    @GET("clinics")            Call<List<Clinic>> getClinics();
    @GET("clinics/{id}")       Call<Clinic>       getClinic(@Path("id") long id);
    @GET("clinics/search")     Call<List<Clinic>> searchClinics(@Query("district") String d);
    @POST("clinics")           Call<Clinic>       createClinic(@Body Clinic c);
    @PUT("clinics/{id}")       Call<Clinic>       updateClinic(@Path("id") long id, @Body Clinic c);
    @DELETE("clinics/{id}")    Call<Void>         deleteClinic(@Path("id") long id);

    // ── Doctors ───────────────────────────────────────────────────────────────
    @GET("doctors")            Call<List<Doctor>> getDoctors();
    @GET("doctors/{id}")       Call<Doctor>       getDoctor(@Path("id") long id);
    @GET("doctors/search")     Call<List<Doctor>> searchDoctors(@Query("specialty") String s, @Query("telemedicine") Boolean t);
    @POST("doctors")           Call<Doctor>       createDoctor(@Body Doctor d);
    @PUT("doctors/{id}")       Call<Doctor>       updateDoctor(@Path("id") long id, @Body Doctor d);
    @DELETE("doctors/{id}")    Call<Void>         deleteDoctor(@Path("id") long id);

    // ── Ambulances ────────────────────────────────────────────────────────────
    @GET("ambulances")         Call<List<Ambulance>> getAmbulances();
    @GET("ambulances/{id}")    Call<Ambulance>       getAmbulance(@Path("id") long id);
    @GET("ambulances/search")  Call<List<Ambulance>> searchAmbulances(@Query("district") String d, @Query("availableOnly") boolean a);
    @POST("ambulances")        Call<Ambulance>       createAmbulance(@Body Ambulance a);
    @PUT("ambulances/{id}")    Call<Ambulance>       updateAmbulance(@Path("id") long id, @Body Ambulance a);
    @DELETE("ambulances/{id}") Call<Void>            deleteAmbulance(@Path("id") long id);

    // ── Pharmacies ────────────────────────────────────────────────────────────
    @GET("pharmacies")         Call<List<Pharmacy>> getPharmacies();
    @GET("pharmacies/{id}")    Call<Pharmacy>       getPharmacy(@Path("id") long id);
    @GET("pharmacies/search")  Call<List<Pharmacy>> searchPharmacies(@Query("district") String d, @Query("open24h") boolean o);
    @POST("pharmacies")        Call<Pharmacy>       createPharmacy(@Body Pharmacy p);
    @PUT("pharmacies/{id}")    Call<Pharmacy>       updatePharmacy(@Path("id") long id, @Body Pharmacy p);
    @DELETE("pharmacies/{id}") Call<Void>           deletePharmacy(@Path("id") long id);

    // ── Diagnostic Centers ────────────────────────────────────────────────────
    @GET("diagnostics")        Call<List<DiagnosticCenter>> getDiagnostics();
    @GET("diagnostics/{id}")   Call<DiagnosticCenter>       getDiagnostic(@Path("id") long id);
    @GET("diagnostics/search") Call<List<DiagnosticCenter>> searchDiagnostics(@Query("district") String d, @Query("test") String t);
    @POST("diagnostics")       Call<DiagnosticCenter>       createDiagnostic(@Body DiagnosticCenter d);
    @PUT("diagnostics/{id}")   Call<DiagnosticCenter>       updateDiagnostic(@Path("id") long id, @Body DiagnosticCenter d);
    @DELETE("diagnostics/{id}")Call<Void>                   deleteDiagnostic(@Path("id") long id);

    // ── Blood Banks ───────────────────────────────────────────────────────────
    @GET("blood-banks")        Call<List<BloodBank>> getBloodBanks();
    @GET("blood-banks/{id}")   Call<BloodBank>       getBloodBank(@Path("id") long id);
    @GET("blood-banks/search") Call<List<BloodBank>> searchBloodBanks(@Query("district") String d);
    @POST("blood-banks")       Call<BloodBank>       createBloodBank(@Body BloodBank b);
    @PUT("blood-banks/{id}")   Call<BloodBank>       updateBloodBank(@Path("id") long id, @Body BloodBank b);
    @DELETE("blood-banks/{id}")Call<Void>            deleteBloodBank(@Path("id") long id);

    // ── Blood Organizations ───────────────────────────────────────────────────
    @GET("blood-organizations")          Call<List<BloodOrganization>> getBloodOrgs();
    @GET("blood-organizations/{id}")     Call<BloodOrganization>       getBloodOrg(@Path("id") long id);
    @GET("blood-organizations/search")   Call<List<BloodOrganization>> searchBloodOrgs(@Query("district") String d);
    @POST("blood-organizations")         Call<BloodOrganization>       createBloodOrg(@Body BloodOrganization o);
    @PUT("blood-organizations/{id}")     Call<BloodOrganization>       updateBloodOrg(@Path("id") long id, @Body BloodOrganization o);
    @DELETE("blood-organizations/{id}")  Call<Void>                    deleteBloodOrg(@Path("id") long id);

    // ── Appointments ──────────────────────────────────────────────────────────
    @GET("appointments/my")      Call<List<Appointment>> myAppointments();
    @GET("appointments/{id}")    Call<Appointment>       getAppointment(@Path("id") long id);
    @POST("appointments")        Call<Appointment>       bookAppointment(@Body Appointment a);
    @PUT("appointments/{id}")    Call<Appointment>       updateAppointment(@Path("id") long id, @Body Appointment a);
    @DELETE("appointments/{id}") Call<Void>              cancelAppointment(@Path("id") long id);

    // ── Telemedicine ──────────────────────────────────────────────────────────
    @GET("telemedicine/my")      Call<List<TelemedicineSession>> myTelemedicine();
    @GET("telemedicine/{id}")    Call<TelemedicineSession>       getTelemedicine(@Path("id") long id);
    @POST("telemedicine")        Call<TelemedicineSession>       createTelemedicine(@Body TelemedicineSession s);
    @PUT("telemedicine/{id}")    Call<TelemedicineSession>       updateTelemedicine(@Path("id") long id, @Body TelemedicineSession s);
    @DELETE("telemedicine/{id}") Call<Void>                      cancelTelemedicine(@Path("id") long id);

    // ── Medicine Reminders ────────────────────────────────────────────────────
    @GET("reminders/my")         Call<List<MedicineReminder>> myReminders();
    @GET("reminders/my/active")  Call<List<MedicineReminder>> myActiveReminders();
    @GET("reminders/{id}")       Call<MedicineReminder>       getReminder(@Path("id") long id);
    @POST("reminders")           Call<MedicineReminder>       createReminder(@Body MedicineReminder r);
    @PUT("reminders/{id}")       Call<MedicineReminder>       updateReminder(@Path("id") long id, @Body MedicineReminder r);
    @DELETE("reminders/{id}")    Call<Void>                   deleteReminder(@Path("id") long id);

    // ── Health Records ────────────────────────────────────────────────────────
    @GET("health-records/my")        Call<List<HealthRecord>> myHealthRecords();
    @GET("health-records/{id}")      Call<HealthRecord>       getHealthRecord(@Path("id") long id);
    @POST("health-records")          Call<HealthRecord>       createHealthRecord(@Body HealthRecord r);
    @PUT("health-records/{id}")      Call<HealthRecord>       updateHealthRecord(@Path("id") long id, @Body HealthRecord r);
    @DELETE("health-records/{id}")   Call<Void>               deleteHealthRecord(@Path("id") long id);

    // ── Approval Requests (user-submitted) ────────────────────────────────────
    @POST("approvals")
    Call<java.util.Map<String, Object>> submitApprovalRequest(@Body java.util.Map<String, Object> body);

    @GET("approvals/my")
    Call<List<java.util.Map<String, Object>>> myApprovals();
}

File: android-app/app/src/main/java/com/lifeplus/healthcare/data/api/ApiService.kt
package com.lifeplus.healthcare.data.api

import com.lifeplus.healthcare.data.model.*
import com.lifeplus.healthcare.model.DashboardSlide
import retrofit2.Response
import retrofit2.http.*

interface ApiService {

    // ── Auth ──────────────────────────────────────────────────────────────────
    @POST("auth/login")
    suspend fun login(@Body body: LoginRequest): Response<AuthResponse>

    @POST("auth/register")
    suspend fun register(@Body body: RegisterRequest): Response<AuthResponse>

    @POST("auth/otp/request")
    suspend fun requestOtp(@Body body: Map<String, String>): Response<Map<String, String>>

    @POST("auth/otp/verify")
    suspend fun verifyOtp(@Body body: Map<String, String>): Response<Map<String, String>>

    @POST("auth/forgot-password/request")
    suspend fun forgotPasswordRequest(@Body body: Map<String, String>): Response<Map<String, String>>

    @POST("auth/forgot-password/reset")
    suspend fun forgotPasswordReset(@Body body: Map<String, String>): Response<Map<String, String>>

    // ── User Profile ──────────────────────────────────────────────────────────
    @GET("users/me")
    suspend fun getMyProfile(): Response<UserProfile>

    @PUT("users/{id}")
    suspend fun updateProfile(@Path("id") id: Long, @Body body: Map<String, String>): Response<UserProfile>

    // ── Hospitals ─────────────────────────────────────────────────────────────
    @GET("hospitals")
    suspend fun getHospitals(): Response<List<Hospital>>

    @GET("hospitals/search")
    suspend fun searchHospitals(@Query("district") district: String): Response<List<Hospital>>

    @GET("hospitals/my")
    suspend fun getMyHospitals(): Response<List<Hospital>>

    @POST("hospitals")
    suspend fun createHospital(@Body hospital: Hospital): Response<Hospital>

    @PUT("hospitals/{id}")
    suspend fun updateHospital(@Path("id") id: Long, @Body hospital: Hospital): Response<Hospital>

    @DELETE("hospitals/{id}")
    suspend fun deleteHospital(@Path("id") id: Long): Response<Unit>

    // ── Clinics ───────────────────────────────────────────────────────────────
    @GET("clinics")
    suspend fun getClinics(): Response<List<Clinic>>

    @GET("clinics/search")
    suspend fun searchClinics(@Query("district") district: String): Response<List<Clinic>>

    @GET("clinics/my")
    suspend fun getMyClinics(): Response<List<Clinic>>

    @POST("clinics")
    suspend fun createClinic(@Body clinic: Clinic): Response<Clinic>

    @PUT("clinics/{id}")
    suspend fun updateClinic(@Path("id") id: Long, @Body clinic: Clinic): Response<Clinic>

    @DELETE("clinics/{id}")
    suspend fun deleteClinic(@Path("id") id: Long): Response<Unit>

    // ── Doctors ───────────────────────────────────────────────────────────────
    @GET("doctors")
    suspend fun getDoctors(): Response<List<Doctor>>

    @GET("doctors/search")
    suspend fun searchDoctors(
        @Query("specialty") specialty: String? = null,
        @Query("hospitalId") hospitalId: Long? = null,
        @Query("telemedicine") telemedicine: Boolean? = null,
        @Query("district") district: String? = null
    ): Response<List<Doctor>>

    @GET("doctors/my")
    suspend fun getMyDoctors(): Response<List<Doctor>>

    @POST("doctors")
    suspend fun createDoctor(@Body doctor: Doctor): Response<Doctor>

    @PUT("doctors/{id}")
    suspend fun updateDoctor(@Path("id") id: Long, @Body doctor: Doctor): Response<Doctor>

    @DELETE("doctors/{id}")
    suspend fun deleteDoctor(@Path("id") id: Long): Response<Unit>

    // ── Ambulances ────────────────────────────────────────────────────────────
    @GET("ambulances")
    suspend fun getAmbulances(): Response<List<Ambulance>>

    @GET("ambulances/search")
    suspend fun searchAmbulances(
        @Query("district") district: String? = null,
        @Query("availableOnly") availableOnly: Boolean = true
    ): Response<List<Ambulance>>

    @GET("ambulances/my")
    suspend fun getMyAmbulances(): Response<List<Ambulance>>

    @POST("ambulances")
    suspend fun createAmbulance(@Body ambulance: Ambulance): Response<Ambulance>

    @PUT("ambulances/{id}")
    suspend fun updateAmbulance(@Path("id") id: Long, @Body ambulance: Ambulance): Response<Ambulance>

    @DELETE("ambulances/{id}")
    suspend fun deleteAmbulance(@Path("id") id: Long): Response<Unit>

    // ── Pharmacies ────────────────────────────────────────────────────────────
    @GET("pharmacies")
    suspend fun getPharmacies(): Response<List<Pharmacy>>

    @GET("pharmacies/search")
    suspend fun searchPharmacies(
        @Query("district") district: String? = null,
        @Query("open24h") open24h: Boolean = false
    ): Response<List<Pharmacy>>

    @GET("pharmacies/my")
    suspend fun getMyPharmacies(): Response<List<Pharmacy>>

    @POST("pharmacies")
    suspend fun createPharmacy(@Body pharmacy: Pharmacy): Response<Pharmacy>

    @PUT("pharmacies/{id}")
    suspend fun updatePharmacy(@Path("id") id: Long, @Body pharmacy: Pharmacy): Response<Pharmacy>

    @DELETE("pharmacies/{id}")
    suspend fun deletePharmacy(@Path("id") id: Long): Response<Unit>

    // ── Blood Banks ───────────────────────────────────────────────────────────
    @GET("blood-banks")
    suspend fun getBloodBanks(): Response<List<BloodBank>>

    @GET("blood-banks/search")
    suspend fun searchBloodBanks(@Query("district") district: String): Response<List<BloodBank>>

    @GET("blood-banks/my")
    suspend fun getMyBloodBanks(): Response<List<BloodBank>>

    @POST("blood-banks")
    suspend fun createBloodBank(@Body bank: BloodBank): Response<BloodBank>

    @PUT("blood-banks/{id}")
    suspend fun updateBloodBank(@Path("id") id: Long, @Body bank: BloodBank): Response<BloodBank>

    @DELETE("blood-banks/{id}")
    suspend fun deleteBloodBank(@Path("id") id: Long): Response<Unit>

    // ── Diagnostic Centers ────────────────────────────────────────────────────
    @GET("diagnostics")
    suspend fun getDiagnostics(): Response<List<DiagnosticCenter>>

    @GET("diagnostics/search")
    suspend fun searchDiagnostics(
        @Query("district") district: String? = null,
        @Query("test") test: String? = null
    ): Response<List<DiagnosticCenter>>

    @GET("diagnostics/my")
    suspend fun getMyDiagnostics(): Response<List<DiagnosticCenter>>

    @POST("diagnostics")
    suspend fun createDiagnostic(@Body diagnostic: DiagnosticCenter): Response<DiagnosticCenter>

    @PUT("diagnostics/{id}")
    suspend fun updateDiagnostic(@Path("id") id: Long, @Body diagnostic: DiagnosticCenter): Response<DiagnosticCenter>

    @DELETE("diagnostics/{id}")
    suspend fun deleteDiagnostic(@Path("id") id: Long): Response<Unit>

    // ── Blood Organizations ───────────────────────────────────────────────────
    @GET("blood-organizations")
    suspend fun getBloodOrgs(): Response<List<BloodOrganization>>

    @GET("blood-organizations/search")
    suspend fun searchBloodOrgs(@Query("district") district: String): Response<List<BloodOrganization>>

    @GET("blood-organizations/my")
    suspend fun getMyBloodOrgs(): Response<List<BloodOrganization>>

    @POST("blood-organizations")
    suspend fun createBloodOrg(@Body org: BloodOrganization): Response<BloodOrganization>

    @PUT("blood-organizations/{id}")
    suspend fun updateBloodOrg(@Path("id") id: Long, @Body org: BloodOrganization): Response<BloodOrganization>

    @DELETE("blood-organizations/{id}")
    suspend fun deleteBloodOrg(@Path("id") id: Long): Response<Unit>

    // ── Donors ────────────────────────────────────────────────────────────────
    @GET("donors")
    suspend fun getDonors(): Response<List<Donor>>

    @GET("donors/search")
    suspend fun searchDonors(
        @Query("bloodGroup") bloodGroup: String? = null,
        @Query("district") district: String? = null
    ): Response<List<Donor>>

    @GET("donors/my")
    suspend fun myDonors(): Response<List<Donor>>

    @POST("donors")
    suspend fun createDonor(@Body donor: Donor): Response<Donor>

    @PUT("donors/{id}")
    suspend fun updateDonor(@Path("id") id: Long, @Body donor: Donor): Response<Donor>

    @DELETE("donors/{id}")
    suspend fun deleteDonor(@Path("id") id: Long): Response<Unit>

    @GET("donors/{id}/eligibility")
    suspend fun checkEligibility(@Path("id") id: Long): Response<Map<String, Any>>

    // ── Blood Requests ────────────────────────────────────────────────────────
    @GET("blood-requests")
    suspend fun getBloodRequests(): Response<List<BloodRequest>>

    @GET("blood-requests/search")
    suspend fun searchBloodRequests(
        @Query("bloodGroup") bloodGroup: String? = null,
        @Query("district") district: String? = null
    ): Response<List<BloodRequest>>

    @GET("blood-requests/my")
    suspend fun myBloodRequests(): Response<List<BloodRequest>>

    @GET("blood-requests/for-donor")
    suspend fun getBloodRequestsForDonor(): Response<List<BloodRequest>>

    @POST("blood-requests")
    suspend fun createBloodRequest(@Body request: BloodRequest): Response<BloodRequest>

    @DELETE("blood-requests/{id}")
    suspend fun deleteBloodRequest(@Path("id") id: Long): Response<Unit>

    // ── Emergency Requests ────────────────────────────────────────────────────
    @POST("emergency-requests")
    suspend fun createEmergencyRequest(@Body request: EmergencyRequest): Response<EmergencyRequest>

    @GET("emergency-requests/my")
    suspend fun myEmergencyRequests(): Response<List<EmergencyRequest>>

    // ── Appointments ──────────────────────────────────────────────────────────
    @GET("appointments/my")
    suspend fun myAppointments(): Response<List<Appointment>>

    @POST("appointments")
    suspend fun bookAppointment(@Body appointment: Appointment): Response<Appointment>

    @DELETE("appointments/{id}")
    suspend fun cancelAppointment(@Path("id") id: Long): Response<Unit>

    // ── Extended Blood Features ───────────────────────────────────────────────
    @PUT("blood-requests/{id}/status")
    suspend fun updateBloodRequestStatus(
        @Path("id") id: Long,
        @Query("status") status: String
    ): Response<BloodRequest>

    @GET("donors/rewards")
    suspend fun getDonorRewards(): Response<Map<String, Any>>

    @POST("donors/verify-donation")
    suspend fun verifyDonation(@Body body: Map<String, Long>): Response<Map<String, Any>>

    @GET("blood-requests/dashboard")
    suspend fun getDashboardBloodRequests(): Response<List<BloodRequest>>

    // ── Telemedicine ──────────────────────────────────────────────────────────
    @GET("telemedicine/my")
    suspend fun myTelemedicine(): Response<List<TelemedicineSession>>

    @POST("telemedicine")
    suspend fun createTelemedicineSession(@Body session: TelemedicineSession): Response<TelemedicineSession>

    // ── Medicine Reminders ────────────────────────────────────────────────────
    @GET("reminders/my")
    suspend fun myReminders(): Response<List<MedicineReminder>>

    @POST("reminders")
    suspend fun createReminder(@Body reminder: MedicineReminder): Response<MedicineReminder>

    @PUT("reminders/{id}")
    suspend fun updateReminder(@Path("id") id: Long, @Body reminder: MedicineReminder): Response<MedicineReminder>

    @DELETE("reminders/{id}")
    suspend fun deleteReminder(@Path("id") id: Long): Response<Unit>

    // ── Health Records ────────────────────────────────────────────────────────
    @GET("health-records/my")
    suspend fun myHealthRecords(): Response<List<HealthRecord>>

    @POST("health-records")
    suspend fun createHealthRecord(@Body record: HealthRecord): Response<HealthRecord>

    @PUT("health-records/{id}")
    suspend fun updateHealthRecord(@Path("id") id: Long, @Body record: HealthRecord): Response<HealthRecord>

    @DELETE("health-records/{id}")
    suspend fun deleteHealthRecord(@Path("id") id: Long): Response<Unit>

    // ── App Settings ──────────────────────────────────────────────────────────
    @GET("settings")
    suspend fun getSettings(): Response<List<AppSetting>>

    @GET("slides")
    suspend fun getDashboardSlides(): Response<List<DashboardSlide>>

    // ── Notifications ─────────────────────────────────────────────────────────
    @GET("notifications/my")
    suspend fun getNotifications(): Response<List<Notification>>

    // ── Reviews ───────────────────────────────────────────────────────────────
    @GET("reviews")
    suspend fun getReviews(@Query("type") type: String, @Query("id") id: Long): Response<List<Review>>

    @POST("reviews")
    suspend fun submitReview(@Body review: Review): Response<Review>

    // ── Chat ──────────────────────────────────────────────────────────────────
    @GET("chat/rooms")
    suspend fun getRooms(): Response<List<ChatRoom>>

    @GET("chat/messages/{roomId}")
    suspend fun getMessages(@Path("roomId") roomId: Long): Response<List<ChatMessage>>

    @POST("chat/send")
    suspend fun sendMessage(@Body msg: ChatMessage): Response<ChatMessage>

    @POST("chat/start/{userId}")
    suspend fun startChat(@Path("userId") userId: Long): Response<ChatRoom>
}
